lib/compat/wordpress-6.2/class-wp-theme-json-resolver-6-2.php

```diff
@@ -32,4 +32,29 @@ public static function theme_has_support() {
 		return wp_theme_has_theme_json();
 	}
 
+	/**
+	 * Private method to clean the cached data after an upgrade.
+	 *
+	 * It is hooked into the `upgrader_process_complete` action.
+	 *
+	 * @see default-filters.php
+	 *
+	 * @param WP_Upgrader $upgrader WP_Upgrader instance.
+	 * @param array       $options  Array of bulk item update data.
+	 */
+	public static function _clean_cached_data_upon_upgrading( $upgrader, $options ) {
```

Comment:
Let stop adding static methods to this class. They are not in part of the WordPress coding standards and are harder to test.

Reply:
I've done a quick search for this pattern and there are multiple cases where this pattern is used:

- [core upgrader](https://github.com/WordPress/wordpress-develop/blob/d2f420493d87465c323190892d4e7a6e4c6070a7/src/wp-admin/includes/class-core-upgrader.php#L273)
- [language pack upgrader](https://github.com/WordPress/wordpress-develop/blob/d2f420493d87465c323190892d4e7a6e4c6070a7/src/wp-admin/includes/class-language-pack-upgrader.php#L49)
- [community events](https://github.com/WordPress/wordpress-develop/blob/d2f420493d87465c323190892d4e7a6e4c6070a7/src/wp-admin/includes/class-wp-community-events.php#L241)
- [privacy policy](https://github.com/WordPress/wordpress-develop/blob/d2f420493d87465c323190892d4e7a6e4c6070a7/src/wp-admin/includes/class-wp-privacy-policy-content.php#L36)
- [WP_Theme](https://github.com/WordPress/wordpress-develop/blob/d2f420493d87465c323190892d4e7a6e4c6070a7/src/wp-includes/class-wp-theme.php#L1546)

I won't list them all, but these should serve to highlight that there are different sensibilities around this. Given this code is architected this way, and it is a very common pattern, I don't think we should do it differently.